pdm/termui.py

```diff
@@ -213,24 +212,23 @@ def logging(self, type_: str = "install") -> Iterator[logging.Logger]:
         """A context manager that opens a file for logging when verbosity is NORMAL or
         print to the stdout otherwise.
         """
-        file_name = mktemp(".log", f"pdm-{type_}-")
-
+        file_name: str | None = None
         if self.verbosity >= Verbosity.DETAIL:
-            handler: logging.Handler = RichHandler(
-                console=_err_console, show_time=False, show_level=False, show_path=False
-            )
```

Comment:
I guess this is a bug in rich?
```
        width (int, optional): The width of the terminal. Leave as default to auto-detect width.
```
it seems like it was failing to auto-detect the width when run from docker :raised_eyebrow: 

StreamHandler (which uses stderr) seems safer. :+1: from me. Thanks so much FrostMing!

Reply:
Yes, RichHandler is designed for TTY usage. It wraps or crops text based on the terminal width, and fills the line with spaces(#1179 ). Changing it to a normal stream handler can easily solve it.